src/domain/LogoutViewModel.ts

```diff
@@ -16,10 +16,11 @@ limitations under the License.
 
 import {Options, ViewModel} from "./ViewModel";
 import {Client} from "../matrix/Client.js";
+import {SegmentType} from "./navigation/index";
 
 type LogoutOptions = { sessionId: string; } & Options;
 
-export class LogoutViewModel extends ViewModel<LogoutOptions> {
+export class LogoutViewModel extends ViewModel<SegmentType, LogoutOptions> {
```

Comment:
Would it make sense to swap the order of the generic params in ViewModel? I think within Hydrogen, once everything is TS-ified, every view model will specify the options type, but I _think_ none will explicitly set the segment type? IIRC, that would avoid have to the above in most view models (once TS).

Reply:
```ts
export class ViewModel<N extends object = SegmentType, O extends Options<N> = Options<N>> extends EventEmitter<{change: never}>
```
Not sure if it can be done that way since the Options generic is dependent on the first one.